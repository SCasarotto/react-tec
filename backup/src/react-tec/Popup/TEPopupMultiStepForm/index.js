import TEPopupMultiStepForm from './TEPopupMultiStepForm'

export default TEPopupMultiStepForm
