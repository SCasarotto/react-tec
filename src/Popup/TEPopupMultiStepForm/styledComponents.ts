import styled from 'styled-components'

import { TEPopup } from '../TEPopup'

export const Popup = styled(TEPopup)`
	.TEPopupContent {
		width: 90%;
		max-width: 900px;
		margin: 100px auto;
		padding: 30px;
	}
`
