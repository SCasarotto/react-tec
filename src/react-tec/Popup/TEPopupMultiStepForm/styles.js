// import { colors } from './../../../config/styles'

export default {
	content: {
		width: '90%',
		maxWidth: '900px',
		margin: '100px auto',
	},
	container: {
		padding: 30,
	},
}
