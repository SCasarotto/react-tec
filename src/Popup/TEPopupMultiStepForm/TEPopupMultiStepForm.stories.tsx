import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEPopupMultiStepForm, TEPopupMultiStepFormProps } from './TEPopupMultiStepForm'

export default {
	title: 'Popup/TEPopupMultiStepForm',
	component: TEPopupMultiStepForm,
	argTypes: {},
} as Meta

const Template: Story<TEPopupMultiStepFormProps> = (args) => <TEPopupMultiStepForm {...args} />

export const Default = Template.bind({})
Default.args = {}
