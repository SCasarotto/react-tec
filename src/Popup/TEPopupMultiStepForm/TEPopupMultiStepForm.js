//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import TEMultiStepForm from './../../Form/TEMultiStepForm'

import { Popup } from './styledComponents'

const TEPopupMultiStepForm = (props) => {
	const handleSubmit = (e) => {
		e.preventDefault()
		const { onSubmit } = props
		onSubmit()
	}
	const { visible, onClose, className, ...rest } = props

	return (
		<Popup visible={visible} className={className}>
			<TEMultiStepForm
				onSubmit={handleSubmit}
				handleCancelOnClick={onClose}
				className='TEPopupMultiStepFormMultiStepForm'
				{...rest}
			/>
		</Popup>
	)
}

TEPopupMultiStepForm.propTypes = {
	visible: PropTypes.bool.isRequired,
	onClose: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired,
}

TEPopupMultiStepForm.defaultProps = {
	visible: false,
}

export default TEPopupMultiStepForm
