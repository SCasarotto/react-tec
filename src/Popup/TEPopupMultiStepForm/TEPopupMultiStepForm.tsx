import React from 'react'

import { TEMultiStepForm, TEMultiStepFormProps } from '../../Form/TEMultiStepForm'

import { Popup } from './styledComponents'
import { TEPopupProps } from './../TEPopup'

export interface TEPopupMultiStepFormProps extends TEPopupProps, TEMultiStepFormProps {
	onClose(): void
	onSubmit(): Promise<any>
}
/**
 *
 * TEPopupMultiStepForm is a combination of `TEPopup` and `TEMultiStepForm`. Besides the props below, reference `TEMultiStepForm` for additional props.
 *
 */
export const TEPopupMultiStepForm = React.forwardRef<HTMLFormElement, TEPopupMultiStepFormProps>(
	(props, ref) => {
		const { visible, onClose, className = '', ...rest } = props

		return (
			<Popup visible={visible} className={`TEPopupMultiStepForm ${className}`}>
				<TEMultiStepForm
					handleCancelOnClick={onClose}
					className='TEPopupMultiStepFormMultiStepForm'
					ref={ref}
					{...rest}
				/>
			</Popup>
		)
	},
)

TEPopupMultiStepForm.defaultProps = {
	visible: false,
}
