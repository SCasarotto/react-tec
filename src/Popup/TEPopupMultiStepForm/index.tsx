import React from 'react'

import {
	TEMultiStepForm,
	TEMultiStepFormProps,
} from '../../Form/TEMultiStepForm'

import { Popup } from './styledComponents'
import { TEPopupProps } from 'Popup/TEPopup'

export interface TEPopupMultiStepFormProps
	extends TEPopupProps,
		TEMultiStepFormProps {
	onClose(): void
	onSubmit(): Promise<any>
	visible: boolean //I don't understand why I need this here since it is in TEPopupProps.....
}
export const TEPopupMultiStepForm: React.FC<TEPopupMultiStepFormProps> = (
	props,
) => {
	const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault()
		const { onSubmit } = props
		onSubmit()
	}
	const { visible, onClose, className = '', ...rest } = props

	return (
		<Popup
			visible={visible}
			className={`TEPopupMultiStepForm ${className}`}>
			<TEMultiStepForm
				onSubmit={handleSubmit}
				handleCancelOnClick={onClose}
				className="TEPopupMultiStepFormMultiStepForm"
				{...rest}
			/>
		</Popup>
	)
}

TEPopupMultiStepForm.defaultProps = {
	visible: false,
}
