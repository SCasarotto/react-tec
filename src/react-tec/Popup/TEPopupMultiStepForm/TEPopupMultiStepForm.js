//
//TE Version 0.2.0
//

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'

import TEPopup from './../TEPopup'
import TEMultiStepForm from './../../Form/TEMultiStepForm'

import styles from './styles'

class TEPopupMultiStepForm extends Component {
	handleSubmit = (e) => {
		e.preventDefault()
		const { onSubmit } = this.props
		onSubmit()
	}
	render() {
		const { visible, onClose, ...rest } = this.props

		return (
			<TEPopup visible={visible} contentStyles={styles.content}>
				<TEMultiStepForm
					onSubmit={this.handleSubmit}
					handleCancelOnClick={onClose}
					containerStyles={styles.container}
					{...rest}
				/>
			</TEPopup>
		)
	}
}

TEPopupMultiStepForm.propTypes = {
	visible: PropTypes.bool.isRequired,
	onClose: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired,
}

TEPopupMultiStepForm.defaultProps = {
	visible: false,
}

export default Radium(TEPopupMultiStepForm)
