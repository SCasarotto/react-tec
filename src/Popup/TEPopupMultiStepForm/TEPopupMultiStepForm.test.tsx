import React from 'react'
import { render } from '@testing-library/react'
import { TEPopupMultiStepForm } from './TEPopupMultiStepForm'

describe('TEPopupMultiStepForm', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(
			<TEPopupMultiStepForm
				visible
				stepData={[]}
				onClose={() => {}}
				onSubmit={async () => {}}
			/>,
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
