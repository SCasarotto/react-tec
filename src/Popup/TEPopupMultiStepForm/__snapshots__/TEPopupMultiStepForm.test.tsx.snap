// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEPopupMultiStepForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jrsJCI eHRuBa TEPopup sc-iqAbSa gJGAiq TEPopupMultiStepForm "
  >
    <div
      class="sc-kEqYlL kePiMs TEPopupContent"
    />
  </div>
</DocumentFragment>
`;
